src/wasm-lib/kcl/src/std/revolve.rs

```diff
@@ -25,65 +25,57 @@ pub struct RevolveData {
     #[serde(default)]
     pub angle: Option<f64>,
     /// Axis of revolution.
-    pub axis: RevolveAxis,
+    pub axis: AxisOrEdgeReference,
     /// Tolerance for the revolve operation.
     #[serde(default)]
     pub tolerance: Option<f64>,
 }
 
-/// Axis of revolution or tagged edge.
+/// Axis or tagged edge.
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, ts_rs::TS, JsonSchema)]
 #[ts(export)]
 #[serde(untagged)]
-pub enum RevolveAxis {
-    /// Axis of revolution.
-    Axis(RevolveAxisAndOrigin),
+pub enum AxisOrEdgeReference {
+    /// Axis and origin.
+    Axis(AxisAndOrigin),
     /// Tagged edge.
     Edge(EdgeReference),
 }
 
-/// Axis of revolution.
+/// Axis and origin.
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, ts_rs::TS, JsonSchema)]
 #[ts(export)]
 #[serde(rename_all = "camelCase")]
-pub enum RevolveAxisAndOrigin {
+pub enum AxisAndOrigin {
     /// X-axis.
     #[serde(rename = "X", alias = "x")]
     X,
     /// Y-axis.
     #[serde(rename = "Y", alias = "y")]
     Y,
-    /// Z-axis.
-    #[serde(rename = "Z", alias = "z")]
-    Z,
     /// Flip the X-axis.
     #[serde(rename = "-X", alias = "-x")]
     NegX,
     /// Flip the Y-axis.
     #[serde(rename = "-Y", alias = "-y")]
     NegY,
-    /// Flip the Z-axis.
-    #[serde(rename = "-Z", alias = "-z")]
-    NegZ,
     Custom {
         /// The axis.
-        axis: [f64; 3],
+        axis: [f64; 2],
         /// The origin.
-        origin: [f64; 3],
+        origin: [f64; 2],
     },
 }
 
-impl RevolveAxisAndOrigin {
+impl AxisAndOrigin {
     /// Get the axis and origin.
     pub fn axis_and_origin(&self) -> Result<(kcmc::shared::Point3d<f64>, kcmc::shared::Point3d<LengthUnit>), KclError> {
         let (axis, origin) = match self {
-            RevolveAxisAndOrigin::X => ([1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::Y => ([0.0, 1.0, 0.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::Z => ([0.0, 0.0, 1.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::NegX => ([-1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::NegY => ([0.0, -1.0, 0.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::NegZ => ([0.0, 0.0, -1.0], [0.0, 0.0, 0.0]),
-            RevolveAxisAndOrigin::Custom { axis, origin } => (*axis, *origin),
+            AxisAndOrigin::X => ([1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::Y => ([0.0, 1.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::NegX => ([-1.0, 0.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::NegY => ([0.0, -1.0, 0.0], [0.0, 0.0, 0.0]),
+            AxisAndOrigin::Custom { axis, origin } => ([axis[0], axis[1], 0.0], [origin[0], origin[1], 0.0]),
```

Comment:
```
 Custom {
        /// The axis.
        axis: [f64; 2],
        /// The origin.
        origin: [f64; 2],
    },
```

Why do these have 3 coordinates? Should these just be 2?

Reply:
engine expects three, we are merely fixing user errors

Reply:
nvm this is correct.